1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/KillAura.kt

```diff
@@ -234,12 +234,90 @@ class KillAura : Module() {
 
         currentTarget ?: return
 
-        val (yaw) = RotationUtils.targetRotation ?: return
-        var strafe = event.strafe
-        var forward = event.forward
+        RotationUtils.targetRotation ?: return
+        val dif = ((MathHelper.wrapAngleTo180_float(mc.thePlayer.rotationYaw - RotationUtils.targetRotation.yaw
+                - 23.5f - 135)
+                + 180) / 45).toInt()
+
+        val yaw = MathHelper.wrapAngleTo180_float(RotationUtils.targetRotation.yaw)
+        val strafe = event.strafe
+        val forward = event.forward
         val friction = event.friction
+        var calcForward = 0f
+        var calcStrafe = 0f
+        /*
+            Rotation Dif
+
+            7 \ 0 / 1     +  +  +      +  |  -
+            6   +   2     -- F --      +  S  -
+            5 / 4 \ 3     -  -  -      +  |  -
+            */
+        when (dif) {
+            0 -> {
+                calcForward = forward
+                calcStrafe = strafe
+            }
+            1 -> {
+                calcForward += forward
+                calcStrafe -= forward
+                calcForward += strafe
+                calcStrafe += strafe
+            }
+            2 -> {
+                calcForward = strafe
+                calcStrafe = -forward
+            }
+            3 -> {
+                calcForward -= forward
+                calcStrafe -= forward
+                calcForward += strafe
+                calcStrafe -= strafe
+            }
+            4 -> {
+                calcForward = -forward
+                calcStrafe = -strafe
+            }
+            5 -> {
+                calcForward -= forward
+                calcStrafe += forward
+                calcForward -= strafe
+                calcStrafe -= strafe
+            }
+            6 -> {
+                calcForward = -strafe
+                calcStrafe = forward
+            }
+            7 -> {
+                calcForward += forward
+                calcStrafe += forward
+                calcForward -= strafe
+                calcStrafe += strafe
+            }
+        }
+
+        if (calcForward > 1f) {
+            calcForward *= 0.5f
+        } else if (calcForward < 0.9f && calcForward > 0.3f) {
+            calcForward *= 0.5f
+        }
+        if (calcForward < -1f) {
+            calcForward *= 0.5f
+        } else if (calcForward > -0.9f && calcForward < -0.3f) {
+            calcForward *= 0.5f
+        }
+
+        if (calcStrafe > 1f) {
+            calcStrafe *= 0.5f
+        } else if (calcStrafe < 0.9f && calcStrafe > 0.3f) {
+            calcStrafe *= 0.5f
+        }
+        if (calcStrafe < -1f) {
+            calcStrafe *= 0.5f
+        } else if (calcStrafe > -0.9f && calcStrafe < -0.3f) {
+            calcStrafe *= 0.5f
+        }
```

Comment:
The code should be optimized.

Reply:
I think this can be done when it is merged